tools/json2bin/evo.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+import pathlib
+
+
+import json2b as j2b
+
+
+from consts import (
+    item
+    moves
+)
+
+
+consts.pokemon import (
+    evo_methods
+    species
+)
+
+
+def get_evo_params(method: EvoMethod, evo: list):
+    maybe_param = evo[1]
+    final_param = 0
+    #None of these take an extra parameter
+    if method in set([
+        evo_methods.EvoMethod.NONE,
+        evo_methods.EvoMethod.HAPPINESS,
+        evo_methods.EvoMethod.HAPPINESS_DAY,
+        evo_methods.EvoMethod.HAPPINESS_NIGHT,
+        evo_methods.EvoMethod.TRADE,
+        evo_methods.EvoMethod.ELECTRIC_FIELD,
+        evo_methods.EvoMethod.MOSSY_STONE,
+        evo_methods.EvoMethod.ICY_STONE]):
+        final_param = 0
+    # These all specify a basic integer param
+    elif method in set([
+        evo_methods.EvoMethod.LEVEL_UP,
+        evo_methods.EvoMethod.LEVEL_ATK_GT_DEF,
+        evo_methods.EvoMethod.LEVEL_ATK_EQ_DEF,
+        evo_methods.EvoMethod.LEVEL_ATK_LT_DEF,
+        evo_methods.EvoMethod.LEVEL_PID_LOW,
+        evo_methods.EvoMethod.LEVEL_PID_HIGH,
+        evo_methods.EvoMethod.LEVEL_NINJASK,
+        evo_methods.EvoMethod.LEVEL_SHEDINJA,
+        evo_methods.EvoMethod.LEVEL_MALE,
+        evo_methods.EvoMethod.LEVEL_FEMALE,
+        evo_methods.EvoMethod.BEAUTY]):
+        final_param = maybe_param
+    # These specify an item
+    elif method in set([
+        evo_methods.EvoMethod.TRADE_WITH_ITEM,
+        evo_methods.EvoMethod.USE_ITEM,
+        evo_methods.EvoMethod.USE_ITEM_MALE,
+        evo_methods.EvoMethod.USE_ITEM_FEMALE,
+        evo_methods.EvoMethod.USE_ITEM_DAY,
+        evo_methods.EvoMethod.USE_ITEM_NIGHT]):
+        final_param = item.Item[maybe_param].value
+    elif method == EvoMethod.KNOW_MOVE:
+        final_param = moves.Move[maybe_param].value
+    elif method == EvoMethod.MON_IN_PARTY:
+        final_param = species.Species[maybe_param].value
+
+    return final_param
+
+
+def table_line(evo_method: int, evo_params: int, species: int) -> bytes:
+    binary = bytearray([])
```

Comment:
Generally, you can (and should) use `.extend` instead of `+=` to add to a list.

Reply:
Update made